### PR TITLE
Next Release 3.1

### PR DESCRIPTION
**Pull request by byjg**
* Using WebRequest in the ApiRequester
* Remove URI (dependency from WebRequest)
* Code clean up.
* Change the HttpClient to support Mock HTTP
* Mock example. 
* ApiRequester now supports any PSR7 RequestInterface implementation by using the method `withPsr7Request()`
* Support to parameter in `formData` (for upload files)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,8 +13,9 @@
 [![Scrutinizer Code Quality](https://scrutinizer-ci.com/g/byjg/php-swagger-test/badges/quality-score.png?b=master)](https://scrutinizer-ci.com/g/byjg/php-swagger-test/?branch=master)
 
 
-A set of tools for test your REST calls based on the OpenApi specification using PHPUnit. Currently this library supports 
-the specifications `2.0` and `3.0`.
+A set of tools for test your REST calls based on the OpenApi specification using PHPUnit. 
+Currently, this library supports the OpenApi specifications `2.0` (former swagger) and `3.0`.
+Some features like callbacks and links weren't implemented.
```
Comment (froschdesign) [inline README.md:18]: I think the referencing to components in external JSON files is missing in this list, like described in "[4.7.23 Reference Object](http://spec.openapis.org/oas/v3.0.3#reference-object)" of the specification.
Or did I miss something and the feature is there?

Comment (byjg) [reply README.md:18]: You are right. The features  `4.7.23.3` and `4.7.23.4` are not implemented. It is an interesting missing feature.

Comment (byjg) [reply README.md]: For now, I am documenting that this feature is missing. But in some moment I am planning to implement this :)

File: README.md [hunk 2]
```diff
@@ -202,6 +203,38 @@ $bodyRequestDef = $schema->getRequestParameters($path, $method);
 $bodyRequestDef->match($requestBody);
 ```
 
+## Validate your Specification
+
+PHP Swagger has the class `MockRequester` with exact the same functionalities of `ApiRequester` class. The only
+difference is the `MockRequester` don't need to request to a real endpoint.
+
+This is particularly usefull if you want to check if your OpenApi specification is returning the expected values.
```
Comment (UlrichEckhardt) [inline README.md:210]: "usefull" -> "useful". Other than that, the OpenApi spec doesn't return anything. Suggestion:

"""This is used to validate request and response against your OpenAPI spec without running any server code."""

I like this addition tough!

Comment (byjg) [reply README.md]: Thanks :+1:

**Pull request merged**